fix: map batch operation states correctly

tmetzke opened the pull request:
## Description

The batch operation state in Java client responses must be transformed considering unsupported values. The EnumUtil provided by the client takes care of this automatically.

## Checklist

<!--- Please delete options that are not relevant. Boxes should be checked by reviewer. -->
- [ ] Enable backports when necessary (fex. [for bug fixes](https://github.com/camunda/camunda/blob/main/CONTRIBUTING.md#backporting-changes) or [for CI changes](https://github.com/camunda/camunda/wiki/CI-&-Automation#when-to-backport-ci-changes)).

## Related issues

closes #37289

tmetzke committed:
fix: map batch operation states correctly

The batch operation state in Java client responses must be transformed considering unsupported values.
The EnumUtil provided by the client takes care of this automatically.

georgios-goulos reviewed (approved):
Perhaps it'd be good to add a unit test that explicitly asserts the fix for the scenario that caused test flakiness.

tmetzke committed:
test: ensure resilient enum conversion

Verifies that the Java SDK handles unknown values in batch operation response type and state.

tmetzke commented:
> Perhaps it'd be good to add a unit test that explicitly asserts the fix for the scenario that caused test flakiness.

Good idea, I adjusted the test case to use the "unknown values" enum to ensure we map those correctly using the enum utility.

megglos commented:
/backport
